### PR TITLE
[libmodplug] Patch CMakeLists.txt to fix install paths

### PR DESCRIPTION
**Pull request by itsmattkc**
**Describe the pull request**

- #### What does your PR fix?  
  Implements Konstanty/libmodplug#61 which fixes a bug in the CMakeLists.txt where DLL files are erroneously installed to the `lib` folder rather than `bin`. Currently the `portfile.cmake` attempts to workaround this by manually renaming the files post-installation, but this is clunky (considering the filename has to be hardcoded), and will also fail if the user is installing only a release OR debug build (since both rename commands are hardcoded and CMake will throw a "file not found" error if they don't both exist).

  By patching the CMakeLists.txt, this is solved at the root of the problem. DLLs are correctly installed to `bin` in the first place (while other library/linking files are still correctly installed to `lib`). The CMake documentation recommends this solution towards the end of the "Installing Targets" section here: https://cmake.org/cmake/help/latest/command/install.html#installing-targets

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All triplets are supported. This patch corrects Win32 issues without breaking other platforms.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  I believe it does. In my opinion, this is a very small and readable patch that I believe will be easy to review. I also believe it makes no extraneous/trivial changes and is the code I am using is fairly well-documented.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes.

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by cenit** [comment]
whenever you touch a ports/<port name> folder content, you are updating a port in vcpkg terminology.
So you need to bump port-version in vcpkg.json (or control if old port) and run the command above after committing the first steps

**Comment by itsmattkc** [comment]
Understood, I believe I have done this now.

**Comment by cenit** [comment]
you did everything in one commit and it’s not valid. You should have first committed your patches with the port version bump, then another commit with the modifications after running vcpkg x-add-versions

**Comment by itsmattkc** [comment]
Alright. How is it now?

File: ports/libmodplug/portfile.cmake [hunk 1]
```diff
@@ -23,15 +24,6 @@ vcpkg_install_cmake()
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 
 if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic AND VCPKG_TARGET_IS_WINDOWS)
```
Comment (JackBoosY) [inline ports/libmodplug/portfile.cmake:26]:
```suggestion
```

File: ports/libmodplug/portfile.cmake [hunk 2]
```diff
@@ -23,15 +24,6 @@ vcpkg_install_cmake()
...
-        set(BIN_NAME libmodplug.dll)
-    else()
-        set(BIN_NAME modplug.dll)
-    endif()
-    file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/bin)
-    file(RENAME ${CURRENT_PACKAGES_DIR}/lib/${BIN_NAME} ${CURRENT_PACKAGES_DIR}/bin/${BIN_NAME})
-    file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/debug/bin)
-    file(RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/${BIN_NAME} ${CURRENT_PACKAGES_DIR}/debug/bin/${BIN_NAME})
     vcpkg_copy_pdbs()
 endif()
```
Comment (JackBoosY) [inline ports/libmodplug/portfile.cmake:28]:
```suggestion
vcpkg_copy_pdbs()
```
We don't need the judgment now.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by itsmattkc** [comment]
Made requested changes.

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Normally we make those changes with the port update / port feature / other port bug.
But this LGTM.

**Pull request merged**